src/Core/Utilities/CoreHelpers.cs

```diff
@@ -183,24 +184,31 @@ public static Uri GetUri(string uriString)
         public static Dictionary<string, string> GetQueryParams(string urlString)
         {
             var dict = new Dictionary<string, string>();
-            if (!Uri.TryCreate(urlString, UriKind.Absolute, out var uri) || string.IsNullOrWhiteSpace(uri.Query))
-            {
-                return dict;
-            }
-            var pairs = uri.Query.Substring(1).Split('&');
-            foreach (var pair in pairs)
+            try
             {
-                var parts = pair.Split('=');
-                if (parts.Length < 1)
+                if (!Uri.TryCreate(urlString, UriKind.Absolute, out var uri) || string.IsNullOrWhiteSpace(uri.Query))
                 {
-                    continue;
+                    return dict;
                 }
-                var key = System.Net.WebUtility.UrlDecode(parts[0]).ToLower();
-                if (!dict.ContainsKey(key))
+                var pairs = uri.Query.Substring(1).Split('&');
+                foreach (var pair in pairs)
                 {
-                    dict.Add(key, parts[1] == null ? string.Empty : System.Net.WebUtility.UrlDecode(parts[1]));
+                    var parts = pair.Split('=');
+                    if (parts.Length < 1)
+                    {
+                        continue;
+                    }
+                    var key = System.Net.WebUtility.UrlDecode(parts[0]).ToLower();
+                    if (!dict.ContainsKey(key))
+                    {
+                        dict.Add(key, parts[1] == null ? string.Empty : System.Net.WebUtility.UrlDecode(parts[1]));
+                    }
                 }
             }
+            catch (Exception ex)
+            {
+                LoggerHelper.LogEvenIfCantBeResolved(ex);
+            }
             return dict;
```

Comment:
I wonder if [HttpUtility.ParseQueryString(...) ](https://learn.microsoft.com/en-us/dotnet/api/system.web.httputility.parsequerystring?redirectedfrom=MSDN&view=net-6.0#System_Web_HttpUtility_ParseQueryString_System_String_) would be the same for this.

Just to check that it's specific for query strings so you can't pass the full uri or there may some problems.
```c#
string uriString = "https://www.google.com/search?q=bitwarden&oq=bitwarden&aqs=chrome.0.02&sourceid=chrome&ie=UTF-8";
var queryStringParams = uriString.Split("?").Last();
var queryStringNameValueCollection = HttpUtility.ParseQueryString(queryStringParams);
```

Reply:
Tried it out and it's for query strings, if you pass the full uri it will not parse the first item well.
To be honest i couldn't find the reason why the client wasn't able to parse the query params. Tried different combination to break it but without success. Ended up protecting with a try catch so at least he could see the 'verification codes' screen, and the rotation value default to 30s.